store/sqlstore/group_supplier.go

```diff
@@ -0,0 +1,864 @@
+		return result
+	}
+
+	if err := groupSyncable.IsValid(); err != nil {
+		result.Err = err
+		return result
+	}
+
+	// If updating DeleteAt it can only be to 0
+	if groupSyncable.DeleteAt != retrievedGroupSyncable.DeleteAt && groupSyncable.DeleteAt != 0 {
+		result.Err = model.NewAppError("SqlGroupStore.GroupUpdateGroupSyncable", "model.group.delete_at.app_error", nil, "", http.StatusInternalServerError)
+		return result
+	}
+
+	// Check if no update is required
+	if (retrievedGroupSyncable.AutoAdd == groupSyncable.AutoAdd) && (retrievedGroupSyncable.CanLeave == groupSyncable.CanLeave) && groupSyncable.DeleteAt != 0 {
+		result.Err = model.NewAppError("SqlGroupStore.GroupUpdateGroupSyncable", "store.sql_group.nothing_to_update", nil, "group_id="+groupSyncable.GroupId+", syncable_id="+groupSyncable.SyncableId, http.StatusInternalServerError)
+		return result
+	}
+
+	// Reset these properties, don't update them based on input
+	groupSyncable.CreateAt = retrievedGroupSyncable.CreateAt
+	groupSyncable.UpdateAt = model.GetMillis()
+
+	var rowsAffected int64
+	switch groupSyncable.Type {
+	case model.GroupSyncableTypeTeam:
+		rowsAffected, err = s.GetMaster().Update(groupSyncableToGroupTeam(groupSyncable))
+	case model.GroupSyncableTypeChannel:
+		rowsAffected, err = s.GetMaster().Update(groupSyncableToGroupChannel(groupSyncable))
+	default:
+		model.NewAppError("SqlGroupStore.GroupUpdateGroupSyncable", "model.group.type.app_error", nil, "group_id="+groupSyncable.GroupId+", syncable_id="+groupSyncable.SyncableId+", "+err.Error(), http.StatusInternalServerError)
+		return result
+	}
+
+	if err != nil {
+		result.Err = model.NewAppError("SqlGroupStore.GroupUpdateGroupSyncable", "store.update_error", nil, err.Error(), http.StatusInternalServerError)
+		return result
+	}
+
+	if rowsAffected == 0 {
+		result.Err = model.NewAppError("SqlGroupStore.GroupUpdateGroupSyncable", "store.sql_group.no_rows_affected", nil, "GroupId="+groupSyncable.GroupId+", SyncableId="+groupSyncable.SyncableId+", SyncableType="+groupSyncable.Type.String()+", "+err.Error(), http.StatusInternalServerError)
+		return result
+	}
+
+	result.Data = groupSyncable
+	return result
+}
+
+func (s *SqlSupplier) GroupDeleteGroupSyncable(ctx context.Context, groupID string, syncableID string, syncableType model.GroupSyncableType, hints ...store.LayeredStoreHint) *store.LayeredStoreSupplierResult {
+	result := store.NewSupplierResult()
+
+	if !model.IsValidId(groupID) {
+		result.Err = model.NewAppError("SqlGroupStore.GroupDeleteGroupSyncable", "model.group.id.app_error", nil, "group_id="+groupID, http.StatusBadRequest)
+		return result
+	}
+
+	if !model.IsValidId(syncableID) {
+		result.Err = model.NewAppError("SqlGroupStore.GroupDeleteGroupSyncable", "model.group_syncable.syncable_id.app_error", nil, "group_id="+groupID, http.StatusBadRequest)
+		return result
+	}
+
+	groupSyncable, err := s.getGroupSyncable(groupID, syncableID, syncableType)
```

Comment:
Again, this whole method could collapse to a trivial `DELETE` statement if we didn't try to use the update methods of gorp.